gbus/worker.go

```diff
@@ -473,6 +485,58 @@ func (worker *worker) invokeHandlers(sctx context.Context, handlers []MessageHan
 		))
 }
 
+func (worker *worker) handleDuplicates(message *BusMessage, delivery amqp.Delivery, msgSpecificLogEntry *logrus.Entry) (bool, error) {
+	if worker.delicatePolicy == DeduplicationPolicyNone {
+		return false, nil
+	}
+	duplicate, err := worker.duplicateStore.MessageExists(message.IdempotencyKey)
+	if err != nil {
+		worker.span.LogFields(slog.String("grabbit", "failed processing duplicate"))
+		worker.log().WithError(err).Error("failed checking for existing message")
+		return true, err
```

Comment:
you can't return nil for a bool value so I return true and an error

Reply:
@vladshub I would return the default value of a bool in go which is false

Reply:
@rhinof this way I'm more conservative since I'd rather have this message rejected as a duplicate and go to DLQ or through the error flow then to have processed it twice if I can't ensure deduplication.